File: 3.x.x/crates-utopia/storage/src/acl.rs
/// Amazon S3 canned ACL grants applied to written objects.
///
/// Used by future S3-compatible adapters; local storage ignores this value.
#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
pub enum Acl {
    Private,
    PublicRead,
    PublicReadWrite,
    AuthenticatedRead,
}

impl Acl {
    pub const fn as_str(self) -> &'static str {
        match self {
            Self::Private => "private",
            Self::PublicRead => "public-read",
            Self::PublicReadWrite => "public-read-write",
            Self::AuthenticatedRead => "authenticated-read",
        }
    }
}

File: packages/storage/src/Storage/Acl.php
<?php

declare(strict_types=1);

namespace Utopia\Storage;

/**
 * Amazon S3 canned ACL grants applied to written objects.
 */
enum Acl: string
{
    case Private = 'private';

    case PublicRead = 'public-read';

    case PublicReadWrite = 'public-read-write';

    case AuthenticatedRead = 'authenticated-read';
}
